lib/manager/bundler/locked-version.ts

```diff
@@ -0,0 +1,24 @@
+const DEP_REGEX = '(?<=\\().*(?=\\))';
+export function extractLockFileEntries(lockFileContent: string): any {
+  const gemLock = { deps: [] };
```

Comment:
```suggestion
  const gemLock = new Map<string,string>();
```
I think every dep should only be there once.

Reply:
I found a couple of exceptions where the same dependency with multiple appearances.
What do you suggest here? 
@ViceIce @rarkins 

https://github.com/rails/rails/blob/667b150c20feade7138a9c0a2b0886e32e830801/Gemfile.lock#L242-L245

Reply:
Take the *first* value that is a valid version, e.g satisfies `.isVersion()`.

Reply:
Okay. Also adding a couple of more different gemfile in the test.

Reply:
What is the semantic difference between `.isVersion()` and `isValid()` ?

Reply:
For ruby it is more or less the same, because `isVersion` calls `isValid`

Reply:
https://github.com/renovatebot/renovate/blob/master/lib/versioning/ruby/index.ts